dealer.py

```diff
@@ -0,0 +1,177 @@
+
+import definitions
+
+class Dealer:
+    """Class that represents the Dealer of a BlackJack game
+    Dealer will make a decision on their Blackjack hand according to the Dealer's default strategy (stand on 17 above; hit on 16 below)
+    
+    Parameters:
+
+    dev_mode (bool): Flag that indicates developer mode. In developer mode all debug and print statements will display.
+
+    """
+
+    def __init__(dev_mode=False):
+        Dealer.dev_mode = dev_mode
+    
+    def is_soft(hand):
+        """Checks whether an Ace is contained within the given hand, meaning the current hand is a soft hand
+
+        Parameters:
+
+        hand (list of ints): The hand that the player was given, represented as a list of integers
+
+        Returns:
+
+        bool:Whether given hand is soft or not
+        """
+
+        return 1 in hand
+
+    def is_pair(hand, card_type):
+        """Checks whether the hand is a pair of a specific type
+
+        Parameters:
+
+        hand (list of ints): The hand that the player was given, represented as a list of integers
+
+        card_type (int) : The face value to check whether it appears as pair (i.e. if type=3, then check if hand is pair 3)
+
+        Returns:
+
+        bool:Whether given hand is a pair of specific type
+        """
+
+        # If hand has more than two cards, automatically cannot be a pair
+        if len(hand) > 2:
+            return False
+
+        # Go through each card and check whether all cards match to the specific type
+        for card in hand:
+            if card != card_type:
+                return False
+
+        # If none of aboce failed, means we have a pair of the specific type
+        return True
+        
+
+
+    def hand_total(hand):
+        """Determine the total of a given hand
+
+        Parameters:
+
+        hand (list of ints): The hand that the player was given, represented as a list of integers
+
+        Returns:
+
+        int:Total of current hand
+        """
+        
+        if Dealer.is_soft(hand):
+            total = [0, 0]
+
+            for val in hand:
+                # Case where we encountered an Ace
+                if (val == 1):
+                    total[0] += 11
+                    total[1] += 1
+                else:
+                    total[0] += val
+                    total[1] += val
+            
+            return total
+
+        else:
+            return list(sum(hand))
+    
+
+    def valid_total(totals):
+        """Determine what total to return
+
+        Parameters:
+
+        total (list of ints): The different totals that can represent a hand
+
+        Return:
+
+        int:The proper total from the given totals
+        """
+        if len(totals) == 1:
+            return totals[0]
+        
+        else:
+            curr_max = max(totals)
+            curr_min = min(totals)
+
+            # error case
+            if curr_max > 21 and curr_min > 21:
```

Comment:
Aren't curr_max and curr_min still lists here? Do we mean to check their sums?

Reply:
nvm python list comparison op

Reply:
But I believe a list is being returned rather than an int still

Reply:
max returns the largest item in the given array, so it will be the highest/lowest in the array of totals.

Reply:
Oh I thought total was holding the lists of the hands, looks good!